src/reconcile/rec_write.c

```diff
@@ -1982,6 +1994,29 @@ __rec_leaf_page_max(WT_SESSION_IMPL *session, WT_RECONCILE *r)
 	return (page_size * 2);
 }
 
+#define	WT_REC_MAX_SAVED_UPDATES	100
```

Comment:
@michaelcahill, an alternative might be to track the memory in the saved updates and use that value plus the argument `len` as the comparison against `r->space_avail`.